client.go

```diff
@@ -276,6 +296,29 @@ func NewClient(appID string, appKey string, host string, port int, realm string,
 	return c, nil
 }
 
+// parseError :
+//  non-exportable helper to parse error response codes between handled (loosely expected http status)
+//  and non-handled status codes.
+func parseError(response *http.Response) (*HttpError, error) {
+	httpError := new(HttpError)
+	handledStatus := []int{400, 404, 500}
```

Comment:
So what the code does here is, if it is a handled error in our documentation, I have added json decoding for it because it will be in the correct format. However, if it is not documented (such as a 503) I can't reliably say if it will have a "status" and "message" key or even be JSON at all, so I just parse out the error and create my own data struct with values pulled from the actual http response. If you look at line 315, if the error was not a handled status (IE in our documentation) then I just parse it from the actual response object. The return on 312 will exit the method with a handled error if it is.